src/Elastic.Apm/Model/DbSpanCommon.cs

```diff
@@ -21,43 +37,80 @@ internal static void EndSpan(Span span, IDbCommand dbCommand, TimeSpan? duration
 					Instance = dbCommand.Connection.Database,
 					Type = Database.TypeSql
 				};
-			}
 
-			if (duration.HasValue) span.Duration = duration.Value.TotalMilliseconds;
+				span.Context.Destination =
+					GetDestination(dbCommand.Connection?.ConnectionString, isEmbeddedDb, defaultPort, dbConnectionStringParser);
+			}
 
-			var providerType = dbCommand.Connection.GetType().FullName;
+			span.End();
+		}
 
-			switch (providerType)
+		private static void GetDefaultProperties(string dbConnectionClassName, out string spanSubtype, out bool isEmbeddedDb, out int? defaultPort)
+		{
+			isEmbeddedDb = false;
+			switch (dbConnectionClassName)
 			{
-				case string str when str.ContainsOrdinalIgnoreCase("Sqlite"):
-					span.Subtype = ApiConstants.SubtypeSqLite;
+				case string str when str.ContainsOrdinalIgnoreCase("SQLite"):
+					spanSubtype = ApiConstants.SubtypeSqLite;
+					isEmbeddedDb = true;
+					defaultPort = null;
+					break;
+				case string str when str.ContainsOrdinalIgnoreCase("MySQL"):
+					spanSubtype = ApiConstants.SubtypeMySql;
+					defaultPort = DefaultPorts.MySql;
+					break;
+				case string str when str.ContainsOrdinalIgnoreCase("Oracle"):
+					spanSubtype = ApiConstants.SubtypeOracle;
+					defaultPort = DefaultPorts.Oracle;
 					break;
-				case string str when str.ContainsOrdinalIgnoreCase("SqlConnection"):
-					span.Subtype = ApiConstants.SubtypeMssql;
+				case string str when str.ContainsOrdinalIgnoreCase("Postgre"):
+					spanSubtype = ApiConstants.SubtypePostgreSql;
+					defaultPort = DefaultPorts.PostgreSql;
+					break;
+				case string str when str.ContainsOrdinalIgnoreCase("NpgSQL"):
+					spanSubtype = ApiConstants.SubtypePostgreSql;
+					defaultPort = DefaultPorts.PostgreSql;
+					break;
+				case string str when str.ContainsOrdinalIgnoreCase("Microsoft"):
+					spanSubtype = ApiConstants.SubtypeMssql;
+					defaultPort = DefaultPorts.MsSql;
+					break;
+				case string str when str.ContainsOrdinalIgnoreCase("System.Data.SqlClient"):
+					spanSubtype = ApiConstants.SubtypeMssql;
+					defaultPort = DefaultPorts.MsSql;
 					break;
 				default:
-					span.Subtype = providerType; //TODO, TBD: this is an unknown provider
+					spanSubtype = dbConnectionClassName; //TODO, TBD: this is an unknown provider
+					defaultPort = null;
 					break;
 			}
+		}
 
-			switch (dbCommand.CommandType)
+		private static string GetSpanAction(CommandType dbCommandType)
```

Comment:
As an alternative [`switch expression`](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#switch-expressions) from C# 8. Fortunately, it's possible after #647. It will be shorter and can be used without an additional method. Smth like that:

```csharp
span.Action = dbCommand.CommandType switch
{
    CommandType.Text => ApiConstants.ActionQuery,
    CommandType.StoredProcedure => ApiConstants.ActionExec,
    CommandType.TableDirect => "tabledirect",
    _ => dbCommand.CommandType.ToString()
};
```

Reply:
Done